core/probe.cpp

```diff
@@ -396,6 +398,11 @@ void Probe::delayedInit()
         showInProcessUi();
 }
 
+void Probe::shutdown()
+{
+    delete this;
```

Comment:
The alternative I see for this is connect(qApp, &QCoreApp::aboutToQuit, [](){ delete Probe::instance(); }), but that's Qt5-only.